public/app/features/datasources/settings/DataSourceSettingsPage.tsx

```diff
@@ -140,6 +142,14 @@ export class DataSourceSettingsPage extends PureComponent<Props> {
     );
   }
 
+  renderMissingEditRightsMessage() {
+    return (
+      <Alert aria-label={selectors.pages.DataSource.readOnly} severity="info" title="Missing rights">
+        You are not allowed to modify this data source. Please contact your server admin to update this data source.
+      </Alert>
```

Comment:
Is the `aria-label` used for tests? Maybe switch to use data-testid instead. Please refer to this doc: https://github.com/grafana/grafana/blob/main/contribute/style-guides/e2e.md#aria-labels-vs-data-testid

Reply:
Thanks, I think I'm in this case. And since I don't use it in test, I'll remove it altogether.